libbindgen/src/ir/context.rs

```diff
@@ -1066,3 +1102,38 @@ impl<'ctx, 'gen> Iterator for WhitelistedItemsIter<'ctx, 'gen>
         Some(id)
     }
 }
+
+pub struct RelatedItemsIter<'ctx, 'gen>
+    where 'gen: 'ctx,
+{
+    ctx: &'ctx BindgenContext<'gen>,
+    seen: ItemSet,
+    to_iterate: VecDeque<ItemId>,
+}
+
+impl<'ctx, 'gen> Iterator for RelatedItemsIter<'ctx, 'gen>
+    where 'gen: 'ctx,
+{
+    type Item = ItemId;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let id = match self.to_iterate.pop_front() {
+            None => return None,
+            Some(id) => id,
+        };
+
+        debug_assert!(self.seen.contains(&id));
+        debug_assert!(self.ctx.items.contains_key(&id));
```

Comment:
This is where the `if is_dangling(id) { ... }` snippet that finds the BFS path would go.